packages/axios-extension/src/abap/adt-catalog/generators/ui-service-generator.ts

```diff
@@ -49,42 +66,64 @@ export class UiServiceGenerator extends AdtService {
+    public async validatePackage(pckg: string): Promise<any> {
+        const response = await this.get('/validation', {
+            headers: {
+                Accept: 'application/vnd.sap.adt.validationMessages.v1+xml'
+            },
+            params: {
+                referencedObject: this.bo.uri,
+                package: pckg,
+                checks: 'package'
+            }
+        });
+        return this.parseResponse(response.data);
+    }
+
+    /**
+     * Validate the service content and package before generation.
+     *
+     * @param content - The content to be validated.
+     * @returns The response of the validation.
+     */
+    public async validateContent(content: string): Promise<ValidationResponse> {
+        const response = await this.post('/validation', content, {
             headers: {
                 'Content-Type': 'application/vnd.sap.adt.repository.generator.content.v1+json',
-                Accept: 'application/vnd.sap.adt.repository.generator.v1+json, application/vnd.sap.as+xml;charset=UTF-8;dataname=com.sap.adt.StatusMessage'
+                Accept: 'application/vnd.sap.adt.validationMessages.v1+xml'
             },
             params: {
                 referencedObject: this.bo.uri,
-                corrNr: transport
+                checks: 'package,referencedobject,authorization'
             }
         });
-        // Service binding is in XML format, ready to be used for the subsequent activation and publish.
         const data = this.parseResponse<any>(response.data);
-        return data.objectReferences;
+        return data.validationMessages?.validationMessage;
     }
 
     /**
-     * Lock the service binding. The class should be configured with the uri of the service binding
-     * The uri is returned from the generate method.
+     * Generate the service binding.
+     *
+     * @param content - The content of the service binding.
+     * @param transport - The transport.
+     * @returns The object references.
      */
-    public async lockServiceBinding() {
-        await this.post('', '', {
+    public async generate(content: string, transport: string): Promise<unknown> {
+        const response = await this.post('', content, {
             headers: {
-                Accept: 'application/*,application/vnd.sap.as+xml;charset=UTF-8;dataname=com.sap.adt.lock.result',
-                'x-sap-adt-sessiontype': 'stateful'
+                'Content-Type': 'application/vnd.sap.adt.repository.generator.content.v1+json',
+                Accept: 'application/vnd.sap.adt.repository.generator.v1+json, application/vnd.sap.as+xml;charset=UTF-8;dataname=com.sap.adt.StatusMessage'
             },
             params: {
-                _action: `LOCK`,
-                accessMode: 'MODIFY'
+                referencedObject: this.bo.uri,
+                corrNr: transport
             }
         });
+        return JSON.parse(response.data);
```

Comment:
try catch?

Reply:
I have a catch for errors from the calling generator, this is also similar to the other methods here for service gen. If it's ok I'll follow that pattern here?